test/extensions/filters/http/cdn_loop/config_test.cc

```diff
@@ -41,6 +43,28 @@ TEST(CdnLoopFilterFactoryTest, BlankCdnIdThrows) {
                              EnvoyException, HasSubstr("value length must be at least"));
 }
 
+TEST(CdnLoopFilterFactoryTest, InvalidCdnId) {
+  NiceMock<Server::Configuration::MockFactoryContext> context;
+
+  envoy::extensions::filters::http::cdn_loop::v3alpha::CdnLoopConfig config;
+  config.set_cdn_id("[not-token-or-ip");
+  CdnLoopFilterFactory factory;
+
+  EXPECT_THAT_THROWS_MESSAGE(factory.createFilterFactoryFromProto(config, "stats", context),
+                             EnvoyException, HasSubstr("is not a valid CDN identifier"));
+}
+
+TEST(CdnLoopFilterFactoryTest, InvalidCdnIdNonHeaderWhitespace) {
+  NiceMock<Server::Configuration::MockFactoryContext> context;
+
+  envoy::extensions::filters::http::cdn_loop::v3alpha::CdnLoopConfig config;
+  config.set_cdn_id("\r\n");
```

Comment:
Sorry I didn't bring this up to your attention earlier, but configs can be validated to not contain \r\n\0 via a protoc-gen-validate validation like https://github.com/envoyproxy/envoy/blob/6321e5d95f7e435625d762ea82316b7a9f7071a4/generated_api_shadow/envoy/config/core/v3/health_check.proto#L86

Reply:
All envoy configs go through validation, so a regression test would look something like
https://github.com/envoyproxy/envoy/blob/4b4dc5cae283f5bd2478b345e0caea8b31cef682/test/extensions/filters/http/ext_authz/ext_authz_test.cc#L418

Reply:
I updated the test that relies on proto validation to catch a ProtoValidationException.

Unless I'm missing something, I don't think it makes sense to use the HTTP_HEADER_VALUE proto validator.  The CDN identifier has more stringent requirements than just matching the HTTP_HEADER_VALUE regular expression so we have to call the `parseCdnId` function.  The `parseCdnId` function also handles the values that HTTP_HEADER_VALUE handles, so we're all set there.

Reply:
Ah gotcha! I see in the docs that it must reference RFC 8586. 

This solution makes sense, then -- the constraints are more stringent than to be able to relfect them in the API via a regex, so the throw works.